lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb

```diff
@@ -106,19 +108,27 @@ def custom_build_settings
           #
           # Saves the content the provided path unless the path exists and the contents are exactly the same.
           #
-          # @return [Object] the result of the generator.
-          #
           def update_changed_file(generator, path)
             if path.exist?
-              result = generator.save_as(support_files_temp_dir)
-              unless FileUtils.identical?(support_files_temp_dir, path)
-                FileUtils.mv(support_files_temp_dir, path)
+              if !generator.respond_to?(:generate)
+                # the generator doesn't support generating string contents, so write it to a tmp file and compare
```

Comment:
The one generator that broke this is the `plist` generator - it uses a function in `Xcodeproj` that only accepts a path. I could work on getting that one but I think getting all the other generators to support this is still worth it

Reply:
could use the underlying serializer from nanaimo if you wanted:

```
plist = Nanaimo::Plist.new(hash, :xml)
Nanaimo::Writer::XMLWriter.new(plist, :pretty => true, :output => f, :strict => false).write
```

Reply:
Yup, I thought of that too. Would need to use another StringIO here so it can "write" to it then pull the string out.

I'll go ahead and do that 👍

Reply:
👍